aws-android-sdk-testutils/src/main/java/com/amazonaws/testutils/AWSTestBase.java

```diff
@@ -0,0 +1,432 @@
+import com.amazonaws.testutils.R ;
+
+@RunWith(AndroidJUnit4.class)
+public abstract class AWSTestBase {
+
+
+    public static class JSONConfiguration  {
+        private JSONObject mJSONObject ;
+        public JSONConfiguration(JSONObject mJSONObject) {
+            this.mJSONObject = mJSONObject ;
+        }
+        JSONObject getCredentials() {
+            try {
+                return mJSONObject.getJSONObject("Credentials");
+            }
+            catch (JSONException je) {
+                throw new RuntimeException(
+                        "Failed to get credentails from testconfiguration.json, please check that it is correctly formed.",
+                        je);
+            }
+
+        }
+        JSONObject getPackages()  {
+            try {
+                return mJSONObject.getJSONObject("Packages");
+            }
+            catch (JSONException je) {
+                throw new RuntimeException(
+                        "Failed to get Packages from testconfiguration.json, please check that it is correctly formed.",
+                        je);
+            }
+         }
+        public JSONObject getPackageConfigure(String packageName) {
+            try {
+                return getPackages().getJSONObject(packageName);
+            }
+            catch (JSONException je) {
+                throw new RuntimeException(
+                        "Failed to get package configuration from testconfiguration.json for " + packageName,
+                        je);
+            }
+        }
+
+        public String getAccessKey() {
+            try {
+                return getCredentials().getString("accessKey");
+            }
+            catch (JSONException je) {
+                throw new RuntimeException(
+                        "Failed to get accessKey from testconfiguration.json, please check that it is correctly formed.",
+                        je);
+            }
+        }
+        public String getSecretKey() {
+            try {
+                return getCredentials().getString("secretKey");
+            }
+            catch (JSONException je) {
+                throw new RuntimeException(
+                        "Failed to get secretKey from testconfiguration.json, please check that it is correctly formed.",
+                        je);
+            }
+        }  
+        public String getAccountId() {
+            try {
+                return getCredentials().getString("accountId");
+            }
+            catch (JSONException je) {
+                throw new RuntimeException(
+                        "Failed to get accountId from testconfiguration.json, please check that it is correctly formed.",
+                        je);
+            }
+        }
+    }
+
+
+    /**
+     * An implementation of AWSCredentialProvider that fetches the credentials
+     * from Odin.
+     */
+    static class JSONAWSCredentialProvider implements AWSCredentialsProvider {
```

Comment:
why is this class necessary?

Reply:
An instance of the class will be used to create a AWSCredentialsProviderChain, of course we create AWSCredentials directly. 
Another option is to create AWSMobileClient to instead of AWSCredentials